docs/api.adoc

```diff
@@ -1319,6 +1361,463 @@ Only one value field can be specified.
 
 |===
 
+== Query DSL Input Types
+
+=== QueryDSLInput
+
+Query DSL input type. Only one field must be provided. More details about Query DSL you can find https://developer.enonic.com/docs/xp/next/storage/dsl[here].
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|boolean: <<BooleanDSLExpressionInput>>
+|Boolean DSL expression
+
+|ngram: <<NgramDSLExpressionInput>>
+|Ngram DSL expression
+
+|stemmed: <<StemmedDSLExpressionInput>>
+|Stemmed DSL expression
+
+|fulltext: <<FulltextDSLExpressionInput>>
+|Fulltext DSL expression
+
+|matchAll: <<MatchAllDSLExpressionInput>>
+|MatchAll DSL expression
+
+|pathMatch: <<PathMatchDSLExpressionInput>>
+|PathMatch DSL expression
+
+|range: <<RangeDSLExpressionInput>>
+|Range DSL expression
+
+|term: <<TermDSLExpressionInput>>
+|Term DSL expression
+
+|like: <<LikeDSLExpressionInput>>
+|Like DSL expression
+
+|in: <<InDSLExpressionInput>>
+|In DSL expression
+
+|exists: <<ExistsDSLExpressionInput>>
+|Exists DSL expression
+
+|===
+
+=== BooleanDSLExpressionInput
+
+Boolean DSL expression input type.
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|should: [<<QueryDSLInput>>]
+|The query should appear in the matching document.
+
+|must: [<<QueryDSLInput>>]
+|The query must appear in matching documents and will contribute to the score.
+
+|mustNot: [<<QueryDSLInput>>]
+|The query must not appear in the matching documents
+
+|filter: [<<QueryDSLInput>>]
+|The query must appear in matching documents.
+
+|boost: Float
+|Any query operator result (expression or compound) can be boosted to change the relevance score of the nodes.
+
+|===
+
+=== NgramDSLExpressionInput
+
+Ngram DSL expression input type.
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|fields: [String]!
+|List of fields (propertyPaths) to include in the search.
+
+|query: String!
+|A query string to match field value(s). Support the set of operators.
+
+|operator: [<<DslOperatorType>>]
+|DSL operator. By default, `OR` (any of the words in the query matches).
+
+|===
+
+=== StemmedDSLExpressionInput
+
+Stemmed DSL expression input type.
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|fields: [String]!
+|List of fields (propertyPaths) to include in the search.
+
+|query: String!
+|A query string to match field value(s). Support the set of operators.
+
+|language: String!
+|A query string to match field value(s). Support the set of operators.
+
+|operator: <<DslOperatorType>>
+|DSL operator. By default, `OR` (any of the words in the query matches).
+
+|boost: Float
+|Any query operator result (expression or compound) can be boosted to change the relevance score of the nodes.
+
+|===
+
+=== FulltextDSLExpressionInput
+
+Fulltext DSL expression input type.
+
+==== Fields
+
+|===
+|Field name: Type | Description
+
+|fields: [String]!
+|List of fields (propertyPaths) to include in the search.
+
+|query: String!
+|A query string to match field value(s). Support the set of operators.
```

Comment:
|query: String!
|A query string to match field value(s). Supports the set of https://developer.enonic.com/docs/xp/next/storage/dsl#search_string_operators[operators].